### ISSUE TITLE
[BUG]:  Issues related to inherited role for native elements requiring hierarchy

### ISSUE THREAD

**Issue by shunguoy** [IBMa/equal-access#2275]
### Project

accessibility-checker-engine

### Description

[ARIA 1.3](https://www.w3.org/TR/wai-aria-1.3/) added presentational role Inheritance for native elements requiring a hierarchy (e.g., table, ol, ul) when their roles are none or presentation. The accessibility descendant elements inherit the 'none' role if an explicit role is not defined. This will affect some engine rules (e.g., hierarchical rules). In addition, accessibility checks for such descendants should be ignored. 

### Steps to reproduce

An example test case:

`<div> ul with none role</div>
<ul role="none">
  <li><a href='https://w3c.org'></a></li>
  <li><a role='button' href='https://aria.org' aria-label="aria">ARIA</a></li>
  <li role="marquee"><a href='https://act.org' aria-label="act">ACT Rules</a></li>
</ul>

<hr style="margin-top:50px;">
<div> table with none role</div>
<table role="none">
  <tbody>
    <tr>
      <td>ACT Rules</td>
      <td>WCAG</td>
    </tr>
    <tr>
      <td>ARIA</td>
      <td>CSS</td>
    </tr>
  </tbody>
</table>`
